From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Praschl <praschlpaul@g-p.at>
Date: Wed, 22 Jul 2026 12:48:26 +0200
Subject: [PATCH 030/117] arm64: dts: apple: add J172 battery and USB power

The generic T8010 nodes are disabled until a board describes their
wiring, leaving J172 without battery, USB-source or charger devices.

Enable UART5 and I2C1 with J172 pinctrl and recovery data, place the
BQ27545 below the UART-backed HDQ bus, and instantiate the Bellatrix and
D2365 supplies. The gauge keeps its monitored-battery description but
does not create a charger supplier link that the W1 family-add path
cannot defer and retry. The shared iPad 6 PMIC node only gains a label
for the J172 extension.

Signed-off-by: Paul Praschl <praschlpaul@g-p.at>

Orchard-replay-of: 961a2743a27e6ab93018c09986fe3a9f7dbee959
---
 arch/arm64/boot/dts/apple/t8010-ipad6.dtsi |  2 +-
 arch/arm64/boot/dts/apple/t8010-ipad7.dtsi | 55 ++++++++++++++++++++++
 2 files changed, 56 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/apple/t8010-ipad6.dtsi b/arch/arm64/boot/dts/apple/t8010-ipad6.dtsi
index 4e17a8448..98b3d82b3 100644
--- a/arch/arm64/boot/dts/apple/t8010-ipad6.dtsi
+++ b/arch/arm64/boot/dts/apple/t8010-ipad6.dtsi
@@ -58,7 +58,7 @@
 &i2c0 {
 	status = "okay";
 
-	pmic@3c {
+	d2365_pmic: pmic@3c {
 		compatible = "apple,angel-pmic", "apple,i2c-pmic";
 		reg = <0x3c>;
 		#address-cells = <1>;
diff --git a/arch/arm64/boot/dts/apple/t8010-ipad7.dtsi b/arch/arm64/boot/dts/apple/t8010-ipad7.dtsi
index 5e0184a23..4f2f9dddb 100644
--- a/arch/arm64/boot/dts/apple/t8010-ipad7.dtsi
+++ b/arch/arm64/boot/dts/apple/t8010-ipad7.dtsi
@@ -14,6 +14,11 @@
 #include "t8010-ipad6.dtsi"
 
 / {
+	ipad7_battery: battery {
+		compatible = "simple-battery";
+		charge-full-design-microamp-hours = <8557000>;
+	};
+
 	spi3_nclk: clock-spi3-nclk {
 		compatible = "fixed-clock";
 		#clock-cells = <0>;
@@ -23,12 +28,62 @@
 };
 
 &pinctrl_ap {
+	battery_swi_pins: battery-swi-pins {
+		pinmux = <APPLE_PINMUX(173, 1)>;
+	};
+
+	i2c1_gpio_pins: i2c1-gpio-pins {
+		pinmux = <APPLE_PINMUX(40, 0)>,
+			 <APPLE_PINMUX(39, 0)>;
+	};
+
 	touch_display_sync_active: touch-display-sync-active-pins {
 		pinmux = <APPLE_PINMUX(176, 1)>,
 			 <APPLE_PINMUX(177, 1)>;
 	};
 };
 
+&serial5 {
+	pinctrl-0 = <&battery_swi_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	onewire {
+		compatible = "ti,hdq-uart";
+
+		battery: fuel-gauge {
+			compatible = "ti,bq27545-hdq";
+			monitored-battery = <&ipad7_battery>;
+		};
+	};
+};
+
+&i2c1 {
+	pinctrl-0 = <&i2c1_pins>;
+	pinctrl-1 = <&i2c1_gpio_pins>;
+	pinctrl-names = "default", "gpio";
+	resets = <&ps_i2c1>;
+	scl-gpios = <&pinctrl_ap 40 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&pinctrl_ap 39 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	status = "okay";
+
+	usb_power: usb-power@13 {
+		compatible = "apple,bellatrix";
+		reg = <0x13>;
+		interrupt-parent = <&pinctrl_aop>;
+		interrupts = <13 IRQ_TYPE_LEVEL_LOW>;
+	};
+};
+
+&d2365_pmic {
+	d2365_charger: charger@180 {
+		compatible = "apple,j172-d2365-charger", "apple,d2365-charger";
+		reg = <0x180 0x1e>, <0xa00 0x0d>;
+		reg-names = "status", "control";
+		power-supplies = <&usb_power>;
+	};
+};
+
 &spi3 {
 	cs-gpios = <&pinctrl_ap 85 GPIO_ACTIVE_LOW>;
 	clocks = <&spi3_nclk>;
